pkg/kubelet/volumemanager/reconciler/reconciler_test.go

```diff
@@ -2339,24 +2339,66 @@ func TestSyncStates(t *testing.T) {
 			},
 		},
 		{
-			name: "when reconstruction fails for a volume, volumes should be cleaned up",
+			name: "when reconstruction fails (mount point does not exist) for a volume, volumes should be added in asw",
 			volumePaths: []string{
 				filepath.Join("pod1", "volumes", "fake-plugin", "pvc-abcdef"),
 			},
 			createMountPoint: false,
 			podInfos:         []podInfo{},
 			verifyFunc: func(rcInstance *reconciler, fakePlugin *volumetesting.FakeVolumePlugin) error {
-				return retryWithExponentialBackOff(reconcilerSyncWaitDuration, func() (bool, error) {
-					err := volumetesting.VerifyTearDownCallCount(1, fakePlugin)
-					if err != nil {
-						return false, nil
-					}
-					return true, nil
-				})
```

Comment:
We will still probably need a test case to check that `syncStates` calls `cleanupMounts` when `reconstructVolume()` fails. You need it to fail in other way now that "mount point does not exist" is not an error.

Reply:
Done. please take another look.